moved/src/move_execution/deposited.rs

```diff
@@ -1,31 +1,144 @@
+const U256_LAYOUT: MoveTypeLayout = MoveTypeLayout::U256;
+
+// Topic identifying the event
+// ETHBridgeFinalized(address indexed from, address indexed to, uint256 amount, bytes extraData)
+const ETH_BRIDGE_FINALIZED: B256 = B256::new(hex!(
+    "31b2166ff604fc5672ea5df08a78081d2bc6d746cadce880747f3643d819e83d"
+));
+
 pub(super) fn execute_deposited_transaction(
     tx: &DepositedTx,
     tx_hash: &B256,
     state: &(impl MoveResolver<PartialVMError> + TableResolver),
     genesis_config: &GenesisConfig,
     block_header: HeaderForExecution,
 ) -> crate::Result<TransactionExecutionOutcome> {
+    #[cfg(any(feature = "test-doubles", test))]
+    if tx.data.is_empty() {
+        return direct_mint(tx, tx_hash, state, genesis_config, block_header);
+    }
+
+    let move_vm = create_move_vm()?;
+    let session_id = SessionId::new_from_deposited(tx, tx_hash, genesis_config, block_header);
+    let mut session = create_vm_session(&move_vm, state, session_id);
+    let traversal_storage = TraversalStorage::new();
+    let mut traversal_context = TraversalContext::new(&traversal_storage);
+    // The type of `tx.gas` is essentially `[u64; 1]` so taking the 0th element
+    // is a 1:1 mapping to `u64`.
+    let mut gas_meter = new_gas_meter(genesis_config, tx.gas.as_limbs()[0]);
+
+    let module = ModuleId::new(
+        evm_native::EVM_NATIVE_ADDRESS,
+        evm_native::EVM_NATIVE_MODULE.into(),
+    );
+    let function_name = evm_native::EVM_CALL_FN_NAME;
+    // Unwraps in serialization are safe because the layouts match the types.
+    let args = vec![
+        Value::address(tx.from.to_move_address())
+            .simple_serialize(&ADDRESS_LAYOUT)
+            .unwrap(),
+        Value::address(tx.to.to_move_address())
+            .simple_serialize(&ADDRESS_LAYOUT)
+            .unwrap(),
+        Value::u256(tx.value.to_move_u256())
+            .simple_serialize(&U256_LAYOUT)
+            .unwrap(),
+        Value::vector_u8(tx.data.iter().copied())
+            .simple_serialize(&evm_native::CODE_LAYOUT)
+            .unwrap(),
+    ];
+    let outcome = session
+        .execute_function_bypass_visibility(
+            &module,
+            function_name,
+            Vec::new(),
+            args,
+            &mut gas_meter,
+            &mut traversal_context,
+        )
+        .map_err(Into::into);
+
+    let mint_params = outcome.and_then(|values| {
+        let evm_outcome = evm_native::extract_evm_result(values);
+        if !evm_outcome.is_success {
+            return Err(UserError::DepositFailure(evm_outcome.output));
+        }
+        // TODO: Should consider ERC-20 deposits here too?
+        let bridge_log = evm_outcome
+            .logs
+            .iter()
+            .find(|l| l.topics()[0] == ETH_BRIDGE_FINALIZED)
+            .ok_or(UserError::DepositFailure(evm_outcome.output))?;
+        let dest_address = AccountAddress::new(bridge_log.topics()[2].0);
```

Comment:
nit: Short explanation of event topic structure would be helpful here.

Reply:
Done in https://github.com/MovedNetwork/op-move/pull/173/commits/8f87282c09a8ad513ea7f833fc8f72b29dc0f926